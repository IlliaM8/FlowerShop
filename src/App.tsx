import Layout from "pages/Layout";
import { MainPage } from "pages/MainPage";
import { FC } from "react";
import { Route, Routes } from "react-router";
const App: FC = () => {
  return (
    <Routes>
      <Route path="/" element={<Layout />}>
        <Route index element={<MainPage />} />
        <Route path="catalog" />
        <Route path="delivery" />
        <Route path="about-us" />
        <Route path="contacts" />
        <Route path="FAQ" />
      </Route>
    </Routes>
  );
};

export default App;
