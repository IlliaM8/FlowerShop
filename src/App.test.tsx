import { render, screen, fireEvent } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import App from "./App";
describe("test app", () => {
  test("renders learn react link", () => {
    render(<App />);
    const helloWord = screen.getByText(/Hello/i);
    const btn = screen.getByRole("button");
    const input = screen.getByPlaceholderText(/text/i);
    expect(helloWord).toBeInTheDocument();
    expect(btn).toBeInTheDocument();
  });
  test("awdawd", async () => {
    render(<App />);
    const helloWord = await screen.findByText(/data/i);
    expect(helloWord).toBeInTheDocument();
  });
  test("click", () => {
    render(<App />);
    const btn = screen.getByTestId("btn");
    expect(screen.queryByTestId("toggle")).toBeNull();
    fireEvent.click(btn);
    expect(screen.getByTestId("toggle")).toBeInTheDocument();
  });
  test("input", () => {
    render(<App />);
    const input = screen.getByPlaceholderText(/text/i);
    expect(screen.queryByTestId("input-val")).toContainHTML("");
    fireEvent.input(input, { target: { value: "awdawdawd" } });
    // userEvent.type(input,'12312')
    expect(screen.getByTestId("input-val")).toContainHTML("awdawdawd");
  });
});
