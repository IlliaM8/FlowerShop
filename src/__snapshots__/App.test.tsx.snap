// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<input
  placeholder="text"
/>
`;

exports[`test app awdawd 1`] = `
<input
  placeholder="text"
/>
`;

exports[`test app renders learn react link 1`] = `
<input
  placeholder="text"
/>
`;

exports[`test app renders learn react link 2`] = `
<input
  placeholder="text"
/>
`;
